contribute/internationalization.md

```diff
@@ -22,15 +22,24 @@ Grafana uses the [i18next](https://www.i18next.com/) framework for managing tran
 ```jsx
 import { Trans } from 'app/core/internationalization';
 
+const SearchTitle = ({ term }) => <Trans i18nKey="search-page.results-title">Results for {{ term }}</Trans>;
+```
+
+Prefer using `<Trans />` for JSX children, and `t()` for props and other JavaScript usage.
+
+There may be cases where you need to interpolate variables inside other components in the translation. In this case, you can use the `values` prop to explicitly pass variables to the translation, and reference them as templated strings in the markup. For example:
+
+```jsx
+import { Trans } from 'app/core/internationalization';
+import { Text } from '@grafana/ui';
+
 const SearchTitle = ({ term }) => (
-  <Trans i18nKey="search-page.results-title">
-    Results for <em>{{ term }}</em>
+  <Trans i18nKey="search-page.results-title" values={{ myVariable: term }}>
```

Comment:
I would also document a different approach of putting the variable into its own component. Like

```ts
function SuccessCount({count}) {
  return <Text color="success">{count}</Text>
}
...
return (
 <Trans i18nKey="...">Results for <SuccessCount count={myVariable} />
</Trans>
)
```

As long as you only have a variable in the component, I could see some confumsers prefering this approach.